README.md

```diff
@@ -25,64 +39,124 @@ example or documentation you're looking at just in case there's a mismatch that
+  * It generates a coverage report you can find in `server/build/jacocoHtml/index.html` in addition to the regular report generated by the `test` task.
+
+### End to end testing with Cypress
+
+End to end (E2E) testing involves the whole software stack rather than one part of it. Our E2E tests look at the behavior of both the client and server and how they interact by simulating what a real user would do with it.
+
+We use [Cypress](https://www.cypress.io/) for our end-to-end tests. There are a few ways to run the E2E tests. They area all started from the `client` directory and require the server be running at the same time (`./gradlew run` in the `server` directory).
+
+* `ng e2e` both builds and serves the client and runs through all the Cypress end-to-end tests once.
+* `ng e2e --watch` builds and serves the client but just opens Cypress for you to be able to run the tests you want without closing automatically.
+  * This is the same as running `ng serve` and `npm run cy:open` (or `npx cypress open`) at the same time. If you are already running `ng serve` it will be easier to do this rather than closing it and running `ng e2e`.
 
-* `ng test` runs the client tests once.
-* `ng test --code-coverage` runs the client tests and generates a coverage file you can find in your client directory `client/coverage/client/index.html`.
-Right click on `index.html` and select `Copy path` and paste it into your browser of choice. For Chrome users, you can drag and drop `index.html` onto the tab area of Chrome and it will open it.
-* `npm run e2e` runs end to end tests. What are e2e tests? They are tests that run the real application and simulate user behavior. They assert that the app is running as expected. NOTE: The server (`./gradlew run`) needs to be actively executing for these tests to work!
+The main page of Cypress looks like this: 
 
-There are GitHub Actions set up in your repo for each of the three checks: JUnit tests for the server, Karma tests for the client, and Protractor tests for end-to-end testing. There are badges above that show the status of these checks on the master branch.
+![image](https://user-images.githubusercontent.com/1300395/107994604-9af40300-6fa2-11eb-9caf-35d804281929.png)
+
+You can click on any of the integration test files to run their tests or run them all. When you run a set of tests you will 
+
+![image](https://user-images.githubusercontent.com/1300395/107994642-acd5a600-6fa2-11eb-8b88-1f2879e82848.png)
+
+There are a lot of neat things you can do here like inspect each test and find which selectors to use in the tests you are writing. We encourage you to look through some of the Cypress documentation below.
+
+### GitHub Actions
+
+There are GitHub Actions set up in your repo for each of the three checks: JUnit tests for the server, Karma tests for the client, and Cypress tests for end-to-end testing. There are badges above that show the status of these checks on the master branch.
```

Comment:
Do we need to mention the `ng lint` check here? Should there be a link to someplace that lists all the "rules" so they have a place to go look up what different things mean when they come up?

Reply:
Yeah i want to write a bit more about running ng lint too.